staging/src/k8s.io/apiserver/pkg/audit/policy/checker.go

```diff
@@ -36,9 +36,29 @@ type Checker interface {
 
 // NewChecker creates a new policy checker.
 func NewChecker(policy *audit.Policy) Checker {
+	for i, rule := range policy.Rules {
+		policy.Rules[i].OmitStages = unionStages(policy.OmitStages, rule.OmitStages)
+	}
 	return &policyChecker{*policy}
 }
 
+func unionStages(s1 []audit.Stage, s2 []audit.Stage) []audit.Stage {
```

Comment:
Add fast path?

```
if len(s1) == 0 {
    return s2
}
if len(s2) == 0 {
    return s1
}
```

Reply:
This will be helpful if we have duplicated elements in s1 or s2.

For example:
s1 = nil
s2 = []audit.Stage{"foo", "foo", "bar"}